.gitlab/e2e_testing_deploy/e2e_deploy.yml

```diff
@@ -2,18 +2,25 @@
 # e2e_deploy stage
 # Contains jobs which deploy Agent package to testing repsoitories that are used in e2e tests.
 
-.setup_rpm_signing_key: &setup_rpm_signing_key
-  - printf -- "$($CI_PROJECT_DIR/tools/ci/fetch_secret.sh $RPM_GPG_KEY)" | gpg --import --batch
-  - EXIT="${PIPESTATUS[0]}"; if [ $EXIT -ne 0 ]; then echo "Unable to locate credentials needs gitlab runner restart"; exit $EXIT; fi
-  - RPM_SIGNING_PASSPHRASE=$($CI_PROJECT_DIR/tools/ci/fetch_secret.sh $RPM_SIGNING_PASSPHRASE) || exit $?
-
-.setup_apt_signing_key: &setup_apt_signing_key
-  - printf -- "$($CI_PROJECT_DIR/tools/ci/fetch_secret.sh $DEB_GPG_KEY)" | gpg --import --batch
-  - EXIT="${PIPESTATUS[0]}"; if [ $EXIT -ne 0 ]; then echo "Unable to locate credentials needs gitlab runner restart"; exit $EXIT; fi
-  - APT_SIGNING_KEY_PASSPHRASE=$($CI_PROJECT_DIR/tools/ci/fetch_secret.sh $DEB_SIGNING_PASSPHRASE) || exit $?
-
-.setup_signing_keys_package:
-  &setup_signing_keys_package # Set up prod apt repo to get the datadog-signing-keys package
+.setup_signing: &setup_signing
+  - ${CI_PROJECT_DIR}/tools/ci/retry.sh aws s3 cp s3://binaries-ddbuild-io-prod/dd-source/domains/seceng/sit/apps/artifact-security/vault-gpg-client/gpg-vault/dev/gpg-vault-v1.5.1-linux-amd64.tar.gz .
+  - tar -xzf gpg-vault-v1.5.1-linux-amd64.tar.gz
+  - mv gpg-vault-linux-amd64 /usr/bin/gpg-vault-bin
+
+  # Set up gpg shim
+  - echo '#!/bin/bash' > /usr/bin/gpg
+  - echo 'GPG_VAULT_COMPAT_MODE=true GPG_USER_ID="Datadog Packages" GPG_USER_EMAIL="package@datadoghq.com" /usr/bin/gpg-vault "$@"' >> /usr/bin/gpg
+  - chmod +x /usr/bin/gpg
+
+  # Set up gpg-vault shim
+  - echo '#!/bin/bash' > /usr/bin/gpg-vault
+  - echo 'GPG_USER_ID="Datadog Packages" GPG_USER_EMAIL="package@datadoghq.com" /usr/bin/gpg-vault-bin "$@"' >> /usr/bin/gpg-vault
+  - chmod +x /usr/bin/gpg-vault
+
+# TODO maybe: add an empty "datadog-signing-keys" package to the pool,
```

Comment:
I think `TODO`s and `FIXME`s are fine when they come with an `(owner)`, i.e. without having to `git blame`.